stock_picking_move_package_to_package/tests/test_move_package_to_package.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.tests import TransactionCase
+
+
+class TestStockPicking(TransactionCase):
+    def setUp(self):
+        super(TestStockPicking, self).setUp()
+        company_obj = self.env["res.company"]
+        self.company = company_obj.create({"name": "Company A"})
+        self.user_demo = self.env["res.users"].create(
+            {
+                "login": "firstnametest",
+                "name": "User Demo",
+                "email": "firstnametest@example.org",
+                "groups_id": [
+                    (4, self.env.ref("base.group_user").id),
+                    (4, self.env.ref("stock.group_stock_user").id),
+                ],
+            }
+        )
+        group_stock_multi_locations = self.env.ref("stock.group_stock_multi_locations")
+        self.user_demo.write(
+            {
+                "company_id": self.company.id,
+                "company_ids": [(4, self.company.id)],
+                "groups_id": [(4, group_stock_multi_locations.id, 0)],
+            }
+        )
+        self.StockPicking = self.env["stock.picking"]
+        self.StockPackage = self.env["stock.quant.package"]
+        self.StockQuant = self.env["stock.quant"]
+        self.Product = self.env["product.product"]
+        self.PackageLevel = self.env["stock.package_level"]
+        self.stock_location = (
+            self.env["stock.location"]
+            .sudo()
+            .search(
+                [("name", "=", "Stock"), ("company_id", "=", self.company.id)], limit=1
+            )
+        )
+        self.stock_location_1 = self.env["stock.location"].create(
+            {
+                "name": "New Stock",
+                "location_id": self.stock_location.warehouse_id.view_location_id.id,
+                "company_id": self.company.id,
+            }
+        )
+        self.picking_type = self.stock_location.warehouse_id.int_type_id
+        self.picking_type.write({"show_entire_packs": True})
+
+    def test_move_package_to_package(self):
```

Comment:
We need to test cases for:
- Products that are not tracked by lot
- Products that are tracked by lot/serial number